src/core/reactor.ts

```diff
@@ -1309,24 +1306,27 @@ protected _getFirstReactionOrMutation(): Reaction<any> | undefined {
         }
     }
 
-    private _uncheckedDisconnect<R extends Present, S extends R>(src: IOPort<S>, dst?:IOPort<R>) {
+    private _uncheckedDisconnect<R extends Present, S extends R>(src: IOPort<S>, dst?: IOPort<R>) {
         Log.debug(this, () => "disconnecting " + src + " and " + dst);
         if (dst instanceof IOPort) {
-            // Register receiver for value propagation.
             let writer = dst.asWritable(this._getKey(dst));
             src.getManager(this._getKey(src)).delReceiver
                 (writer as WritablePort<S>);
             this._dependencyGraph.removeEdge(dst, src);
         } else {
-            let nodes = this._dependencyGraph.getEdges(src);
-            for (let node of nodes) {
-                if (node instanceof IOPort) {
-                    let writer = src.asWritable(this._getKey(node));
-                    src.getManager(this._getKey(src)).delReceiver
-                        (writer as WritablePort<S>);
+            let nodes = this._dependencyGraph.getBackEdges(src);
+            if (nodes.size == 0) {
+                throw Error("There are no destinations to disconnect")
```

Comment:
I don't think this is an error. There's just nothing to do, so we can return. Let's not error.

Reply:
Yes, I agree! I fixed the issue.